verilog/CST/parameters.cc

```diff
@@ -77,6 +77,31 @@ const verible::TokenInfo& GetParameterNameToken(const verible::Symbol& symbol) {
   return ABSL_DIE_IF_NULL(identifier_leaf)->get();
 }
 
+const std::vector<const verible::TokenInfo*> GetAllParameterNameTokens(
+    const verible::Symbol& symbol) {
+  //Can be used only in kParamDeclaration context
+  auto t = symbol.Tag();
+  CHECK_EQ(t.kind, verible::SymbolKind::kNode);
+  CHECK_EQ(NodeEnum(t.tag), NodeEnum::kParamDeclaration);
+
+  std::vector<const verible::TokenInfo*> identifiers;
+  identifiers.push_back(&GetParameterNameToken(symbol));
+
+  /* Find any following declarations (look for assignments) */
+  auto assignments = SearchSyntaxTree(symbol, NodekParameterAssign());
+
```

Comment:
> [performance] Instead of searching, can the assignments be directly accessed as subtrees positionally?

What would be the best way to do that if we don't know how many are there? Should I just incrementally check if another one exists and if so add it to the vector?

Reply:
I refactored the code to extract some additional functions. There are no tests for them yet as I am not fully sure I captured what you meant.

I am still not sure how to approach extracting all of the assignments without the matcher, so it would be nice if you could give me some pointers, thanks!